config_template.toml

```diff
@@ -0,0 +1,78 @@
+# cargo-temp's configuration file.
+
+# ================
+# General settings
+# ================
+
+# Directory where temporary projects are created.
+# Default: System cache directory.
+# If the directory does not exist, it will be created automatically.
+temporary_project_dir = "{}"
+
+# Display a welcome message explaining how to exit and preserve the project on when a new temporary
+# project is created.
+# Default: true
+welcome_message = {}
+
+# Prompt for confirmation before deleting the project on exit.
+# Default: false
+# prompt = true
+
+# Directory where the temporary projects will be preserved.
+# Default: None, temporary projects are not moved on exit.
+# preserved_project_dir = "/home/user/lab"
+
+# Version Control System (VCS) to use for new projects..
+# Possible values: "git", "none" or any VCS supported by `cargo new -vcs`.
+# Default: "git" (implicit default from Cargo).
+# vcs = "pijul"
+
+# Number of commits to retain in the history when cloning a Git repository.
+# Set to 0 to keep the full history.
+# Default: 1 (only the latest commit).
+# git_repo_depth = 3
+
+# Override the default target directory for cargo builds.
+# Default: project-specific `target` directory.
+# cargo_target_dir = "/tmp/cargo-target"
+
+# ===============
+# Editor settings
+# ===============
+
+# Path to your preferred editor or IDE.
+# If specified, `cargo-temp` will open the editor instead of a shell.
+
+# Example for VS Code:
+# editor = "{}"
+# editor_args = [ "--wait", "--new-window" ]
+
+# ============
+# Subprocesses
+# ============
+
+# You can spawn subprocesses to run alongside your temporary project.
+#
+# The required fields for each subprocess:
+# - `command`: a shell command to run (e.g., "cargo watch").
+# - `foreground`: Boolean. If true, the process runs in the foreground; if false, it runs in the
+#   background.
+#
+# Optional fields:
+# - `working_dir`: Override the working directory (default: temporary project directory).
+# - `keep_on_exit`: Keep the process running after exiting the shell/editor (default: false).
+#   Note: This setting has no effect for foreground processes.
+#
+# Platform specifics settings:
+# - Unix
+#   - `stdout`: Boolean. Enable or disable stdout output (default: false for background, true for
+#   foreground).
+#   - `stderr`: Boolean. Enable or disable stderr output (default: true for foreground, false for
+#   foreground).
+# - Windows
+#   - `inherit_handles`: Boolean. Allow the subprocess to inherit handles from the parent process
+#   (default: false).
+#
+# [[subprocess]]
+# command = "cargo watch"
```

Comment:
probably not a valid example. it needs to be opened in a new terminal and in background